### PR TITLE
VZ-7869 PSR Network Policy

### PR DESCRIPTION
**Pull request by mbenmclean**
Adds network policy so that workers can access opensearch in clusters with Calico. Also combined the opensearch authorization policy into one file.

### CODE REVIEW

**Review by michael-cico — APPROVED** [review]

**Review by stevshap — APPROVED** [review]

File: tools/psr/manifests/charts/worker/charts/opensearch/templates/authorizationpolicy.yaml [hunk 1]
```diff
@@ -1,6 +1,26 @@
 # Copyright (c) 2022, Oracle and/or its affiliates.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
```
Comment (pfmackin) [inline tools/psr/manifests/charts/worker/charts/opensearch/templates/authorizationpolicy.yaml:5]: Why do you need this change?

File: tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021, 2022, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "worker.fullName" .}}-{{ .Release.Namespace }}-master-netpol
```
Comment (pfmackin) [inline tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml:7]: remove -netpol from the name

File: tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  policyTypes:
+    - Ingress
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: {{ .Release.Namespace }}
+          podSelector:
+            matchLabels:
+              app: psr-worker
```
Comment (pfmackin) [inline tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml:42]: The psr-worker label needs to be added to the worker chart

Comment (mbenmclean) [reply tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml:42]: It is already there

Comment (mbenmclean) [reply tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml:42]: Actually updated the labels

File: tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml (hunk 2 above)
Comment (mbenmclean) [reply tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml]: fixed

File: tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "worker.fullName" .}}-{{ .Release.Namespace }}-ingest-netpol
```
[the remark this answers is not in the document]
Comment (mbenmclean) [reply tools/psr/manifests/charts/worker/charts/opensearch/templates/networkpolicy.yaml]: fixed

**Review by pfmackin — APPROVED** [review]

File: tools/psr/manifests/charts/worker/templates/deployment.yaml [hunk 5]
```diff
@@ -16,7 +16,7 @@ spec:
     metadata:
       labels:
         psr.verrazzano.io/worker: {{ template "worker.fullName" .}}
-        app: psr-{{ .Values.global.envVars.PSR_WORKER_TYPE }}
```
Comment (pfmackin) [inline tools/psr/manifests/charts/worker/templates/deployment.yaml:19]: leave this label and add the new one

**Review by pfmackin — APPROVED** [review]

**Pull request merged**